homeassistant/components/enigma2/media_player.py

```diff
@@ -81,49 +81,44 @@ async def async_setup_platform(
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up of an enigma2 media player."""
-    if discovery_info:
-        # Discovery gives us the streaming service port (8001)
-        # which is not useful as OpenWebif never runs on that port.
-        # So use the default port instead.
-        config[CONF_PORT] = DEFAULT_PORT
-        config[CONF_NAME] = discovery_info["hostname"]
-        config[CONF_HOST] = discovery_info["host"]
-        config[CONF_USERNAME] = DEFAULT_USERNAME
-        config[CONF_PASSWORD] = DEFAULT_PASSWORD
-        config[CONF_SSL] = DEFAULT_SSL
-        config[CONF_USE_CHANNEL_ICON] = DEFAULT_USE_CHANNEL_ICON
-        config[CONF_MAC_ADDRESS] = DEFAULT_MAC_ADDRESS
-        config[CONF_DEEP_STANDBY] = DEFAULT_DEEP_STANDBY
-        config[CONF_SOURCE_BOUQUET] = DEFAULT_SOURCE_BOUQUET
-
-    base_url = URL.build(
-        scheme="https" if config[CONF_SSL] else "http",
-        host=config[CONF_HOST],
-        port=config.get(CONF_PORT),
-        user=config.get(CONF_USERNAME),
-        password=config.get(CONF_PASSWORD),
-    )
 
-    session = async_create_clientsession(hass, verify_ssl=False, base_url=base_url)
+    entry_data = {
+        CONF_HOST: config[CONF_HOST],
+        CONF_PORT: config[CONF_PORT],
+        CONF_USERNAME: config[CONF_USERNAME],
+        CONF_PASSWORD: config[CONF_PASSWORD],
+        CONF_SSL: config[CONF_SSL],
+        CONF_USE_CHANNEL_ICON: config[CONF_USE_CHANNEL_ICON],
+        CONF_DEEP_STANDBY: config[CONF_DEEP_STANDBY],
+        CONF_SOURCE_BOUQUET: config.get(CONF_SOURCE_BOUQUET),
```

Comment:
This is given a default value by the schema:
```suggestion
        CONF_SOURCE_BOUQUET: config[CONF_SOURCE_BOUQUET],
```